### PR TITLE
[buildkite] Add third-party tests to periodic pipeline

### PR DESCRIPTION
**Pull request by brianseeders**
- Bring over third-party test jobs from Jenkins
- Refactor legacy-vault stuff to be more re-usable, since it's used in several places now
- Allow file copies in `GeoIpHttpFixture` to overwrite existing files. This is to avoid `FileAlreadyExistsException` that I got both in Buildkite and on my local machine.
- Fix the `geoip_use_service=true` flag for the geoip tests. They are currently running incorrectly in Jenkins. See #99389

[Tested here](https://buildkite.com/elastic/elasticsearch-periodic/builds/351) (only third-party tests are relevant).

### CODE REVIEW

File: test/fixtures/geoip-fixture/src/main/java/fixture/geoip/GeoIpHttpFixture.java [hunk 1]
```diff
@@ -106,15 +107,25 @@ protected void after() {
     }
 
     private void copyFiles() throws Exception {
-        Files.copy(GeoIpHttpFixture.class.getResourceAsStream("/geoip-fixture/GeoLite2-ASN.tgz"), source.resolve("GeoLite2-ASN.tgz"));
-        Files.copy(GeoIpHttpFixture.class.getResourceAsStream("/geoip-fixture/GeoLite2-City.mmdb"), source.resolve("GeoLite2-City.mmdb"));
+        Files.copy(
```
Comment (mark-vieira) [inline test/fixtures/geoip-fixture/src/main/java/fixture/geoip/GeoIpHttpFixture.java:110]: Why was this not needed before? Are we not using clean working directories for these tests anymore?

Comment (brianseeders) [reply test/fixtures/geoip-fixture/src/main/java/fixture/geoip/GeoIpHttpFixture.java:110]: I honestly don't know why it's not happening in Jenkins, but it happens on my laptop, so it's not exclusive to Buildkite. The buildkite workspace starts with a clean checkout.

Comment (mark-vieira) [reply test/fixtures/geoip-fixture/src/main/java/fixture/geoip/GeoIpHttpFixture.java:110]: Yeah, this strikes me as odd. We should be creating a unique temp directory on every initialization here. If we are overwriting files that means we aren't, or `Files.createTempDirectory` isn't doing what I think it should. My fear is we are using common directories here we risk running into a race condition where one fixture tries to blow away the files of another one mid-test execution.

Comment (brianseeders) [reply test/fixtures/geoip-fixture/src/main/java/fixture/geoip/GeoIpHttpFixture.java:110]: See #99389

**Comment by brianseeders** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.10|[](https://github.com/elastic/elasticsearch/pull/100094)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/100095)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**